packages/react/src/combobox/utils.ts

```diff
@@ -2,7 +2,7 @@ import { useId } from '../core';
 
 export function useComboboxInputId(idProp?: string) {
 	const autoId = useId();
-	return idProp || `combobox-input-${autoId}`;
+	return idProp ?? `combobox-input-${autoId}`;
```

Comment:
I don't think we should do this one. It's now less safe in that you could accidentally pass an empty string to `useComboboxInputId` via a couple of components and have duplicate `id`s on the page. Looks like it's [this flagged issue regarding nullish coalescing](https://sonarcloud.io/project/issues?cleanCodeAttributeCategories=CONSISTENT&impactSoftwareQualities=MAINTAINABILITY&resolved=false&id=agriculturegovau_agds-next&open=AY88Kpvu-1NVyxUQKsi_)
>Prefer using nullish coalescing operator (`??`) instead of a logical or (`||`), as it is a safer operator.

SonarCloud is simply wrong here, it is not a safer operator.

Reply:
Interesting. We are already using the `??` for the same functions dropdown-menu/utils.ts and field/Field.tsx, which is why I thought I'd just go with what Sonar wants and what we have done